### ISSUE TITLE
Syntax checking "unknown to compiler"

### ISSUE THREAD

**Issue by SuperRiderTH** [enigma-dev/enigma-dev#1202]
When trying to run a project, the syntax checking fails in a script with the message: "Unexpected symbol `\': unknown to compiler." This symbol does not appear in the script.

![image](https://user-images.githubusercontent.com/5368844/37489112-fe115d14-286c-11e8-9829-bc778ec4c5de.png)

**Comment by RobertBColton** [comment]
What exactly is on line 127?

**Comment by JoshDreamland** [comment]
That's my question, as well.

**Comment by SuperRiderTH** [comment]
The screenshot shows the end of that script, it ends at line 127, including 0.
All that is there is `action_win              =   30;`

**Comment by RobertBColton** [comment]
Try this again with the 1.8.40 release of LGM I just gave you please.

**Comment by JoshDreamland** [comment]
Sounds as though this problem's LGM-side. Yes?

**Comment by SuperRiderTH** [comment]
This still occurs on 1.8.40.

**Comment by RobertBColton** [comment]
Your thinking is correct @SuperRiderTH but I don't know how or why. Is this a regression? Did your project have this problem before when you were compiling with ENIGMA? If you didn't have the error before then this must be a new issue related to some recent ENIGMA changes.

**Comment by SuperRiderTH** [comment]
I don't remember encountering this error before, I try to compile my big GM:S project sometimes in ENIGMA to see how it handles it, but this parsing error is new. It's been a while, but the last issue I had with this was a segfault happening later that I referenced in #1055.

**Comment by SuperRiderTH** [comment]
Well the line changed to 94, and that is literally just an empty line.

**Comment by RobertBColton** [comment]
Ok, I was just asking so I know if we can say this is a regression. I guess this could possibly be the string related errors still. Try and make a copy of this project, get rid of all the other objects and stuff (don't forget LGM allows multiple select with the `Shift` key), and narrow the problem down to just that single object. If you still get the error, try copying that code to your clipboard and pasting it into an empty project. We need to know how that goes so we can see if it's reproducible.

**Comment by RobertBColton** [comment]
@SuperRiderTH if you can it would be helpful to also know your `gcc -v` since the string errors are on newer the gcc versions typically. Definitely a JDI bug though because I see `\` has no results in this script.

**Comment by SuperRiderTH** [comment]
Whew, been a bit. This is still causing a problem.
```
sean@Noire:~$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.3.0-16ubuntu3' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --with-as=/usr/bin/x86_64-linux-gnu-as --with-ld=/usr/bin/x86_64-linux-gnu-ld --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.3.0 (Ubuntu 7.3.0-16ubuntu3) ```

**Comment by JoshDreamland** [comment]
Huh. That's just sad. Anyway, please add `#undef minor` to your definitions file (under ENIGMA settings). Some genius decided it would be a good idea to define that in `sysmacros.h` (POSIX header).

**Comment by SuperRiderTH** [comment]
Adding ```#undef minor``` to the definitions file makes it proceed further, but it still runs into 

```Check `obj_GameController::create...Syntax error in object `obj_GameController', Create event:0:
Line 402, position 2 (absolute 11918): Unexpected symbol `\': unknown to compiler```

Create event 0 is 27 lines long.

**Comment by SuperRiderTH** [comment]
Adding ```#undef major``` allows it to continue, I get to writing events and get a fatal error.
```Writing events
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f5501c33014, pid=4780, tid=4866
#
# JRE version: Java(TM) SE Runtime Environment (10.0.2+13) (build 10.0.2+13)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (10.0.2+13, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libstdc++.so.6+0x12b014]  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const+0x14
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P" (or dumping to /home/sean/.ENIGMA/enigma-dev/core.4780)
#
# An error report file with more information is saved as:
# /home/sean/.ENIGMA/enigma-dev/hs_err_pid4780.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

[hs_err_pid4780.log](https://github.com/enigma-dev/enigma-dev/files/2260951/hs_err_pid4780.log)

**Comment by JoshDreamland** [comment]
Which event is being written when that crash occurs?

**Comment by SuperRiderTH** [comment]
It is at the start of writing events.

```
Request for type of self: returning int!
New level: 0
code from here: e=2.25;};
Writing events
#
# A fatal error has been detected by the Java Runtime Environment:
```

**Comment by SuperRiderTH** [comment]
Sure, but I'm not sure what would be considered the first event. Here is the create event from the obj_GameController object. They were in different actions, but I threw them all here.

https://pastebin.com/cYdRC3qB

**Comment by RobertBColton** [comment]
I am no longer getting these build errors with the new JDI that #1659 just merged. I downloaded the zip with the `player_init` script that supposedly caused this error, and it compiled and ran just fine. I also copied the create event into an object and placed said object in a room and that compiled and ran fine too only after I commented `show_debug_layer` out since we don't have that function yet.

**Comment by RobertBColton** [comment]
@SuperRiderTH considering the above, and the fact that we've not only merged a newer JDI that better handles macros, but also moved the POSIX stuff around recently, can you try updating to the latest master and give this another go?

**Comment by RobertBColton** [comment]
Also #1677 was merged so now we only use enigma_user for compilation. All of the above may have seriously resolved this issue, but I can't tell unless you try. I'm going to give the ticket another week or so before automatically closing it as resolved.

**Comment by SuperRiderTH** [comment]
![Screenshot from 2019-05-20 21-42-59](https://user-images.githubusercontent.com/5368844/58062426-5676d180-7b48-11e9-9c6d-08bbb8829e32.png)

Looks like it is still an issue on Linux

**Comment by RobertBColton** [comment]
Huh, ok, so one in the bag we haven't resolved yet. Alright, I'll keep this open and update when anything comes about that may resolve it.
